### PR TITLE
Feature/dataset: Dataset creation

### PR DESCRIPTION
**Pull request by evaplaysviolin**
Created a base set of pairs (10 words and their translations in French)
Wrote preliminary function for generating dataset from raw data
Output for `datasetMap` should be the Map we will work with for game play?

Blockers:

- Looking for review on whether this is anywhere near correct approach

### CODE REVIEW

File: src/VocabGameData.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  ["yes", "oui"],
+  ["no", "non"],
+  ["can you help me, please?", "pouvez-vous m’aider, s’il vous plaît?"]
+];
+
+let dataset = [];
+function createDataset(base) {
+  base.forEach(function(array) {
+    dataset.push([
+      array[0],
```
Comment (serpient) [inline src/VocabGameData.js:18]: are you trying to use pure arrays, or a MAP object here?

Comment (serpient) [reply src/VocabGameData.js:18]: oh i see, you are making it an array first and then converting to a MAP object at the end.

Comment (evaplaysviolin) [reply src/VocabGameData.js:18]: Yes, but this isn't any particular decision I made, it's because I didn't know if I could do it directly with MAP

**Review by serpient — COMMENTED** [review]
your code looks good -- 
im going into this branch and push some tests to make sure everything is as expected,
then we can probably approve.

File: src/VocabGameData.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        translation: array[0],
+        isSelected: null,
+        isCorrect: null
+      }
+    ]);
+  });
+}
+
+createDataset(basePairs);
+let datasetMap = new Map(dataset);
```
Comment (serpient) [inline src/VocabGameData.js]: ok, I would tuck in the new Map() conversion within the function `createDataset`

so then all we have  to do is run the function, and it spits out the Map dataset without doing another step.

Comment (serpient) [reply src/VocabGameData.js]: let me know when u made the change and ill pull!

Comment (evaplaysviolin) [reply src/VocabGameData.js]: @serpient I made the changes!

File: src/VocabGameData.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  ["thank you", "merci"],
+  ["thank you very much", "merci beaucoup"],
+  ["you're welcome", "de rien"],
+  ["excuse me", "excusez-moi"],
+  ["yes", "oui"],
+  ["no", "non"],
+  ["can you help me, please?", "pouvez-vous m’aider, s’il vous plaît?"]
+];
+
+let datasetMap;
```
Comment (serpient) [inline src/VocabGameData.js]: this doesnt need to be defined in global scope as its only used in the function

File: src/VocabGameData.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      array[1],
+      {
+        word: array[1],
+        translation: array[0],
+        isSelected: null,
+        isCorrect: null
+      }
+    ]);
+  });
+  datasetMap = new Map(dataset);
```
Comment (serpient) [inline src/VocabGameData.js]: you can just `return new Map(dataset)` here and not keep track of datasetMap at all.

essentially, you are using datasetMap just to return the value, so technically, its not necessary.

also i think you forgot to return the dataset on this line.

**Review by serpient — CHANGES_REQUESTED** [review]
just a few little things!

File: src/VocabGameData.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+createDataset(basePairs);
```
Comment (serpient) [inline src/VocabGameData.js]: i mean here, we would define it like so
` export const datasetMap = createDataset(basePairs) `

so we initialize this variable here and export it at the same time.

Comment (evaplaysviolin) [reply src/VocabGameData.js]: Okay, I think I understand, let me make these changes

**Comment by serpient** [comment]
to run tests, you can run 
`npm test --watchAll` in a different terminal window

**Review by serpient — APPROVED** [review]

**Comment by serpient** [comment]
@evaplaysviolin 

these are the Error handling cases your function should handle (not sure if you get the notifications like I do in VSCODE, possibly this is the Jest extension)
image

**Pull request merged**